src/widgets/PieChart.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2018
+ */
+
+import React, { useState, useEffect } from 'react';
+import { View, StyleSheet } from 'react-native';
+import PropTypes from 'prop-types';
+import { VictoryLabel, VictoryPie } from 'victory-native';
+import { APP_FONT_FAMILY, GREY } from '../globalStyles';
+
+/**
+ * A charting widget for displaying Pie chart report.
+ *
+ * @prop  {string}       id      The report ID.
+ * @prop  {string}       title   The title of the report.
+ * @prop  {string}       type    The type of chart to use to display the report,
+ *                               options are BarChart, LineChart and PieChart.
+ * @prop  {data}         array   An array of {x, y} datapoints to plot.
+ * @prop  {width}        number  The width of the parent container.
+ * @prop  {height}       number  The height of the parent continer.
+ */
+
+export const ReportChart = ({ report: initialReport }, props) => {
+  const [report, setReport] = useState(initialReport);
+  const [width, setWidth] = useState(null);
+  const [height, setHeight] = useState(null);
+
+  // Same affect as componentDidUpdate
+  useEffect(() => {
+    if (report.id !== props.report.id) setReport(props.report);
+  });
+
+  // Victory Native sizes are set using absolute values. Parents dimensions are used to
+  // calculate relative values for width and height for each chart.
+  const onLayout = event => {
+    setWidth(event.nativeEvent.layout.width);
+    setHeight(event.nativeEvent.layout.height);
```

Comment:
I think keep these in an object `{ height: x, width: y }`

In almost every use case, we will update these at the same time. But, when we call the function that is returned from `useState` (in the case `setWidth` or `setHeight`) this will cause the function to re-render. So, we would be re-render twice!

Reply:
Done here: 0f3f2a7
Also not 100% sure about this, maybe there's a better way to do it: 
```
const onLayout = event => {
    const newStateObj = {
      height: event.nativeEvent.layout.width,
      width: event.nativeEvent.layout.height,
    };
    setState(newStateObj);
  };
```